Feat/http2

linonetwo opened the pull request:
closes #8017

I had to expose some internal of server to test it is really running http2.

linonetwo committed:
feat: basic http2 h2c

linonetwo committed:
test: allow test http2 internal performance

linonetwo committed:
docs: time

linonetwo committed:
fix: review

linonetwo committed:
lint

linonetwo committed:
refactor: use "connection-tracking": "yes" in test

linonetwo committed:
refactor: reduce test callback hell indentation by using async await

linonetwo committed:
Create #9405.tid

linonetwo committed:
release note

linonetwo committed:
lint: remove useless check

linonetwo committed:
github-links

pmario commented on core-server/server/server.js:
```diff
@@ -16,7 +16,15 @@ if($tw.node) {
+		http2;
+	// Try to load http2 module (available in Node.js 8.4+)
+	try {
+		http2 = require("http2");
```
I personally would suggest the oldest Node.js version we can support is a [LTS which has not reached end of live](https://nodejs.org/en/about/previous-releases). Today: 2025 Nov. 12 this is LTS v20.19.5 which is in maintenance mode till April 2026. 

So currently supported are: LTS 20, 22 and 24. IMO we do have no resources to deal with other versions than LTS. Just my thoughts.

linonetwo committed:
Enhance HTTP/2 documentation and reverse proxy section

Updated HTTP/2 support details and added reverse proxy usage instructions.

linonetwo committed:
Enhance h2c parameter documentation with examples

Added configuration examples and relationship details for h2c parameter.

linonetwo committed:
Revise HTTP/2 parameter documentation for version 5.4.0

Updated the HTTP/2 web server parameter documentation to reflect changes in versioning and added a reference to 'Using HTTP/2'.

linonetwo committed:
Merge branch 'TiddlyWiki:master' into feat/http2

linonetwo commented:
It takes so long, long enough for me to by more servers, and setup an k3s network, so I have higress-ingress as reverse proxy to add http2 and gzip and cache-control.
